sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,372 @@
+                    (not issparse(data[0]), "Sparse input is not supported."),
+                    (not model_is_sparse, "Sparse coefficients are not supported."),
+                ]
+            )
+
+            return patching_status
+
+        def _onedal_gpu_supported(self, method_name, *data):
+            patching_status = PatchingConditionsChain(
+                f"sklearn.linear_model.{self.__class__.__name__}.fit"
+            )
+
+            if method_name == "fit":
+                alpha_is_scalar = _is_numeric_scalar(self.alpha)
+                patching_status.and_condition(
+                    alpha_is_scalar,
+                    "Non-scalar alpha is not supported for GPU.",
+                )
+
+                return self._onedal_fit_supported(patching_status, method_name, *data)
+
+            if method_name in ["predict", "score"]:
+                return self._onedal_predict_supported(patching_status, method_name, *data)
+
+            raise RuntimeError(
+                f"Unknown method {method_name} in {self.__class__.__name__}"
+            )
+
+        def _onedal_cpu_supported(self, method_name, *data):
+            patching_status = PatchingConditionsChain(
+                f"sklearn.linear_model.{self.__class__.__name__}.fit"
+            )
+
+            if method_name == "fit":
+                return self._onedal_fit_supported(patching_status, method_name, *data)
+
+            if method_name in ["predict", "score"]:
+                return self._onedal_predict_supported(patching_status, method_name, *data)
+
+            raise RuntimeError(
+                f"Unknown method {method_name} in {self.__class__.__name__}"
+            )
+
+        def _initialize_onedal_estimator(self):
+            onedal_params = {
+                "fit_intercept": self.fit_intercept,
+                "alpha": self.alpha,
+                "copy_X": self.copy_X,
+            }
+            self._onedal_estimator = onedal_Ridge(**onedal_params)
+
+        def _daal_fit(self, X, y, sample_weight=None):
+            daal4py_fit_ridge(self, X, y, sample_weight)
+            self._onedal_estimator.n_features_in_ = _num_features(X, fallback_1d=True)
+            self._onedal_estimator.coef_ = self.coef_
+            self._onedal_estimator.intercept_ = self.intercept_
+
+        def _onedal_fit(self, X, y, sample_weight, queue=None):
+            # `Sample weight` is not supported. Expected to be None value.
+            assert sample_weight is None
+
+            check_params = {
+                "X": X,
+                "y": y,
+                "dtype": [np.float64, np.float32],
+                "accept_sparse": ["csr", "csc", "coo"],
+                "y_numeric": True,
+                "multi_output": True,
+            }
+            if sklearn_check_version("1.2"):
+                X, y = self._validate_data(**check_params)
+                self._validate_params()
```

Comment:
move to before dispatch (in fit()) to match other estimators

Reply:
https://github.com/intel/scikit-learn-intelex/pull/1843#discussion_r1645012243
valid point from @icfaust (if i understood it correctly), let me know what you think

Reply:
Thanks